### PR TITLE
Fix SDL wrong success response to split RPCs

### PR DESCRIPTION
**Pull request by pvvasilev**
The following RPCs are split into several HMI
requests. If the result for UI is `UNSUPPORTED_RESOURCE` but
one of the other requests is `SUCCESS`, SDL
should respond with `UNSUPPORTED_RESOURCE, success:true`

The changed RPCs are:
DeleteCommand
PerformInteraction
ChangeRegistration
SetGlobalProperties
PerformAudioPassThru
Alert

### CODE REVIEW

**Comment by pvvasilev** [comment]
Please review @AKalinich-Luxoft @Kozoriz @anosach-luxoft @VProdanov

File: src/components/application_manager/src/commands/mobile/alert_request.cc [hunk 1]
```diff
@@ -230,6 +230,9 @@ bool AlertRequest::PrepareResponseParameters(
   result_code = PrepareResultCodeForResponse(ui_alert_info, tts_alert_info);
   info = MergeInfos(
       ui_alert_info, ui_response_info_, tts_alert_info, tts_response_info_);
+  if (ui_alert_info.is_unsupported_resource && tts_alert_info.is_ok) {
```
Comment (anosach-luxoft) [inline src/components/application_manager/src/commands/mobile/alert_request.cc]: @pvvasilev we have `PrepareResultCodeForResponse()` method for such manipulations. Why don't you use it?

Comment (VProdanov) [inline src/components/application_manager/src/commands/mobile/alert_request.cc]: @pvvasilev maybe 

if (!result && tts_alert_info.is_ok && ui_alert_info.is_unsupported_resource) {
   result = true;
}

for consistency with the code around..

Comment (pvvasilev) [inline src/components/application_manager/src/commands/mobile/alert_request.cc]: @anosach-luxoft Changed in cbabd4bdcba73c8595b5da9763cd79fb4aa8b77d

Comment (pvvasilev) [inline src/components/application_manager/src/commands/mobile/alert_request.cc]: @VProdanov Changed in cbabd4bdcba73c8595b5da9763cd79fb4aa8b77d

**Comment by pvvasilev** [comment]
Rebuild required

**Comment by pvvasilev** [comment]
Rebuild required

**Comment by jacobkeeler** [comment]
This behavior doesn't seem correct, is there any documentation which states that this is the correct behavior?

**Comment by jacobkeeler** [comment]
Closing for now, feel free to reopen if you find documentation stating that this is the correct behavior

**Pull request closed without merging**